core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/GroupUniqueIDDomainResolver.java

```diff
@@ -91,26 +100,37 @@ public String getDomainForGroupId(String groupId, int tenantId) throws UserStore
 
             // Read the cache first.
             String domainName = uniqueIdDomainCache.get(groupId);
-            if (domainName != null) {
+
+            if (org.apache.commons.lang.StringUtils.isBlank(domainName)) {
```

Comment:
Can we check and change the existing carbon StringUtils import to apache StringUtils

Reply:
Fixed: 6c08cef558b386ba44af7d826510ac5d53840a43